### PR TITLE
Storybook Environment construction

### PR DESCRIPTION
**Pull request by WULCAN**
## Description
This PR updates the usage of Environment constructor in .storybook/preview to the constructor's new signature.

To reduce the risk of future mismatched arguments, I converted preview.js to a tsx file.

## Notes to reviewer
```
yarn storybook
```
Open the story titled PersonDetailsCard and check the browser console.

Open the deployment Preview - gen3-storybook, and test there too.

## Related
This relates to PR #2130

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -43,6 +43,7 @@
     "next-env.d.ts",
     "**/*.ts",
     "**/*.tsx",
+    ".storybook/preview.tsx",
```
Comment (richardolsson) [inline tsconfig.json:46]: Having renamed this file will cause major conflicts with #2111, but it's not a big file so I'm sure we'll figure it out when we get there.

File: .storybook/preview.tsx [hunk 2]
```diff
@@ -61,15 +62,14 @@ class MockApiClient extends FetchApiClient {
...
     <ThemeProvider theme={theme}>
       <Story />
     </ThemeProvider>
   ),
-  (Story) => {
+  (Story: StoryFn) => {
     const store = createStore();
-    const router = useRouter();
-    const env = new Environment(store, new MockApiClient(), router);
+    const env = new Environment(new MockApiClient());
```
Comment (richardolsson) [inline .storybook/preview.tsx:72]: Very good catch!

**Review by richardolsson — APPROVED** [review]
Nice work catching this. I have looked at the code and tried the preview build. As far as I can tell it's ready for merge. 💯

File: tsconfig.json (hunk 1 above)
Comment (WULCAN) [reply tsconfig.json:46]: It turned out to not be so bad. git merge understood the file was renamed and there was only a minor conflict to resolve in `.storybook/preview`. See #2358

Comment (richardolsson) [reply tsconfig.json:46]: Git, I underestimated you. 🙏

**Pull request merged**